neuro_san/internals/run_context/langchain/llms/standard_langchain_llm_factory.py

```diff
@@ -89,27 +110,32 @@ def create_base_chat_model(self, config: Dict[str, Any]) -> LangChainLlmResource
                                                           module_name="langchain_openai.chat_models.base",
                                                           install_if_missing="langchain-openai")
 
-            # Our run-time model resource here is httpx client which we need to control directly:
-            openai_proxy = self.get_value_or_env(config, "openai_organization", "OPENAI_PROXY")
-            request_timeout = config.get("request_timeout")
-            http_client = httpx.AsyncClient(proxy=openai_proxy, timeout=request_timeout)
-
-            # Now build an OpenAI client using this resource:
-            async_openai_client: AsyncOpenAI = AsyncOpenAI(
-                api_key=self.get_value_or_env(config, "openai_api_key",
-                                              "OPENAI_API_KEY"),
-                base_url=self.get_value_or_env(config, "openai_api_base", "OPENAI_API_BASE"),
-                organization=self.get_value_or_env(config, "openai_organization", "OPENAI_ORG_ID"),
-                timeout=request_timeout,
-                max_retries=config.get("max_retries"),
-                http_client=http_client
-            )
+            # See if there is an async_client to be had from the llm_client passed in
+            async_client: Any = None
+            if llm_client is not None:
+                async_openai_client = llm_client.get_client()
+                if async_openai_client is not None:
+                    # Necessary reach-in.
+                    async_client = async_openai_client.chat.completions
 
             # Now construct LLM chat model we will be using:
             llm = ChatOpenAI(
-                async_client=async_openai_client.chat.completions,
+                async_client=async_client,
                 model_name=model_name,
                 temperature=config.get("temperature"),
+
+                # This next group of params should always be None when we have async_client
+                openai_api_key=self.get_value_or_env(config, "openai_api_key",
+                                                     "OPENAI_API_KEY", async_client),
+                openai_api_base=self.get_value_or_env(config, "openai_api_base",
+                                                      "OPENAI_API_BASE", async_client),
+                openai_organization=self.get_value_or_env(config, "openai_organization",
+                                                          "OPENAI_ORG_ID", async_client),
+                openai_proxy=self.get_value_or_env(config, "openai_organization",
+                                                   "OPENAI_PROXY", async_client),
+                request_timeout=self.get_value_or_env(config, "request_timeout", None, async_client),
+                max_retries=self.get_value_or_env(config, "max_retries", None, async_client),
```

Comment:
Restore the client-oriented constructor values.
The new get_value_or_env() with the async_client passed in will take care of None values if we already have a client.